### ISSUE TITLE
Transaction signing fails when UserProfile is restored from disk

### ISSUE THREAD

**Issue by Isabello** [iov-one/bcp-redux#161]
We use `writer.signAndCommit` which uses a reference to profile. When we replace the profile, this reference is never updated. This leads to errors like this.

```
userprofile.js:158 Uncaught (in promise) Error: Entry of id ed25519-simpleaddress:a54511c47dd69d1a50254a149fdf8dc483a66e2b854a5c648092b746f097aa8a does not exist in keyring
    at UserProfile.entryInPrimaryKeyring (userprofile.js:158)
    at UserProfile.eval (userprofile.js:125)
    at Generator.next (<anonymous>)
    at eval (userprofile.js:7)
    at new Promise (<anonymous>)
    at __awaiter (userprofile.js:3)
    at UserProfile.signTransaction (userprofile.js:124)
    at IovWriter.eval (writer.js:54)
    at Generator.next (<anonymous>)
    at fulfilled (writer.js:4)
```

We need to disconnect existing writers and reinitialize with the new profile, as there is no way to update this reference after the writer is constructed.

**Comment by Isabello** [comment]
For future reference:

We should refactor the transaction signing to use the user profile's `signTransaction` function: https://iov-one.github.io/iov-core-docs/latest/iov-core/classes/userprofile.html#signtransaction
It should also submit this transaction using `IovWriter`'s commit functionality (note this does not exist at this time): https://iov-one.github.io/iov-core-docs/latest/iov-core/classes/iovwriter.html **Blocked

Give that the above is blocked, we should atleast provide a "reboot" function for existing writers.

**Comment by rudi-cilibrasi** [comment]
https://github.com/iov-one/bcp-redux/blob/master/src/redprom/reset.mjs#L55 is close

**Comment by Isabello** [comment]
This is still an issue. Due to the way the load from disk works, we have to replace the existing userprofile instance in the store, as we dont load into an existing profile, it requires a new construction.

**Comment by ethanfrey** [comment]
You would load in the boot process, before connecting the chains, right?
I think we just need to update the writer creation.

**Comment by Isabello** [comment]
Originally the boot stages were introduced to allow this for initial startup. But if the app is already running we have to reset the writer to use the new profile. Essentially running boot stage 2 again.

**Comment by ethanfrey** [comment]
why load profile from disk in the middle of the app?
i mean, yeah, it is like a hard reset doable.
but what is the usecase?

**Comment by Isabello** [comment]
Immediate use case I can think of:

I am a user with multiple persisted profiles, I like to have a profile/backup for each of my keys separately.

**Comment by ethanfrey** [comment]
So you "logout" and close one profile completely (along with the telescope pointing to appropriate address), and then "login" opening a new profile and reinitializing the view.

Sounds like a good case for `reset` to clean up and redo boot with a different data store

**Comment by rudi-cilibrasi** [comment]
Yes the only reason I was ever pushing the reset action cleanup sequence including the fully elaborated one that includes close connections, reset redux state, and reboot was to allow arbitrary reconfiguration via dialog that allows the view peer to construct an edit peer that is then modified by the user and confirmed via an accept button or discarded via cancel. Only in the case where approve is hit would we then (I thought) save the new configuration then call the performCloseResetAndReboot.  Figured it should cover many common cases but maybe not all.
